File: src/money-management-service/Core/Pagination.cs
namespace money_management_service.Core
{
    public class Pagination
    {
        public int Page { get; set; }

        public int Size { get; set; }

        public int Total { get; set; }

        public Pagination() {
            Page = 0;
            Size = 0;
            Total = 0;
        }

        public Pagination(int Page, int Size, int Total)
        {
            this.Page = Page;
            this.Size = Size;
            this.Total = Total;
        }
    }
}

File: src/money-management-fe/src/core/components/pagination/pagination.ts
import { Component, computed, input, Input, signal } from '@angular/core';

@Component({
  selector: 'mon-pagination',
  imports: [],
  templateUrl: './pagination.html',
  styleUrl: './pagination.css'
})
export class Pagination {
  size = input<number>(10);
  page = input<number>(0);
  total = input<number>(0);


  curentPage = signal(0);

  listPage = computed(() => {
    const result: any[] = [];

    const totalPage = Math.ceil(this.total() / this.size());
    if (totalPage <= 9) {
      for(let i = 1; i <= totalPage; ++i) {
        result.push(i);
      }
    }

    else {
      
    }


    return result;
  });
}
